arch.mk

```diff
@@ -1030,9 +1034,12 @@ ifeq ($(TARGET),sim)
   LD_END_GROUP=
   BOOT_IMG=test-app/image.elf
   CFLAGS+=-DARCH_SIM
+  CFLAGS+=-DWOLFBOOT_USE_STDLIBC
+  LDFLAGS +=-Wl,-gc-sections -Wl,-Map=wolfboot.map
```

Comment:
Adding these LDFLAGS breaks my ability to build and run the simulator on the Mac. Could you gate around USE_GCC perhaps?

Reply:
I guess you are referring to the -DWOLFBOOT_USE_STDLIBC. I've gated that behind USE_GCC as suggested.

Reply:
It was `LDFLAGS +=-Wl,-gc-sections -Wl,-Map=wolfboot.map` causing issue with MacOS.